### DISCUSSION TITLE
Requires an access token key

### DISCUSSION THREAD

**Discussion by ousmorez** [rdilweb/docusaurus-plugin-remote-content#54] [Q&A]
To connect to some repositories like gitlab or private github repositories we need an access token key to use it into the config file.
But I didn't find any option like that.
Can you help me?

**Comment by RDIL** [comment]
You should be able to use the `requestConfig` option.

**Reply by SalamHH** [reply]
Could you please provide a small example on how to specify the access token key with the requestConfig option? I am not able to make it work with private repos :(

**Reply by RDIL** [reply]
What provider are you using for the private repositories?  GitHub?  And what content do you need to fetch (markdown, images, etc)?

**Reply by SalamHH** [reply]
Yes i am using github, and only need to fetch markdown. All i want is a small example code of the requestConfig option.

**Comment by Daxilus** [comment]
I was trying to get a readme file from a private gitlab repository as a test but every time I start from my local docusaurus it saves an html file from the gitlab login page.

   'docusaurus-plugin-remote-content',
    {
      name: "content", 
      sourceBaseUrl: 'https://gitlab.com/group/project/-/raw/master', 
      outDir: "docs/", // the base directory to output to.
      documents: ["README.md"], // the file names to download
      requestConfig: {
        method: 'get',
        responseType: "arraybuffer",
        headers: {
          "PRIVATE-TOKEN": XXXXXX
        }
      }
    }
Could you tell me what I'm doing wrong here?

**Reply by RDIL** [reply]
You may need to use GitLab's repository files api: https://docs.gitlab.com/ee/api/repository_files.html#get-file-from-repository

**Reply by lukasPan** [reply]
Hi,
can you please show how did you set up the config for GitLab repository? 
I tried it but it is still doesn't work. Not sure what set to `sourceBaseUrl` because according official [gitlab doc](https://docs.gitlab.com/ee/api/repository_files.html#get-file-from-repository) you need set up the `ref` on the end of url.

**Comment by TimRoe** [comment]
Just as an FYI for anyone finding this topic in the future: not sure if it's a Docusaurus configuration option, but when adding an auth token for a private repo to pull documentation into a public repo with this plugin and attempting to deploy it to GitHub pages, GitHub flagged it as attempting to expose the auth token. 

Sure enough: looking the assets/js/main.#######.js file that gets generated as part of the Docusaurus build and is bundled into the deployed website, it appears to include a post-processed version of the Docusaurus config file--so despite having the auth token as an Environment secret for the repo/GitHub Action and accessing via `process.env.AUTH_TOKEN` in the code, it was exposing the actual auth token in a file that was to be deployed to the internet.

Fortunately, the private repo is being made public soon so it's whatever to just manually propagate over the doc and automate it later, but figured I'd give a heads up that, at least with how our Docusaurus/build process is configured, it may be exposing your private repo auth token.
